core/src/main/scala/org/apache/spark/resource/ResourceUtils.scala

```diff
@@ -476,7 +476,7 @@ private[spark] object ResourceUtils extends Logging {
       if (maxTaskPerExec < (execAmount * numParts / taskAmount)) {
         val origTaskAmount = treq.amount
         val taskReqStr = s"${origTaskAmount}/${numParts}"
-        val resourceNumSlots = Math.floor(execAmount * numParts / taskAmount).toInt
+        val resourceNumSlots = Math.floor(execAmount * numParts / taskAmount.toDouble).toInt
```

Comment:
Before this pr, the result here was a Long value from `execAmount * numParts / taskAmount`. Is there a need to perform `Math.floor` on it? @tgravescs Do you have time to help check this logic? Thanks

Reply:
it doesn't looks like the floor is needed here. its longs and ints, the other functions called before this take care of doubles.

Reply:
Changed it!

Reply:
Thanks @tgravescs